scripts/util/resolve-types.js

```diff
@@ -50,18 +50,34 @@ module.exports = (shapesWithoutNames) => {
     });
   };
 
+  const isEnumOf = pattern => key => (
+    key.enum && key.enum.length &&
+    key.enum.every(pattern.test.bind(pattern))
+  );
+
+  const isEnumOfFloats = isEnumOf(/\d+(\.\d+)/);
+
   resolve.map = (sh) => {
     const key = resolve.shape(sh.key);
     if (key.type !== 'String' && !key.enum) {
       console.warn(`Unexpected map key type ${key.type}, don't know how to decode`);
       process.exit(1);
     }
     const value = resolve.shape(sh.value);
-    return render.nothing({
-      type: `(Dict ${key.type} ${value.type})`,
-      decoder: `(${jsonDecode}.dict ${value.decoder})`,
-      extraImports: ['import Dict exposing (Dict)'],
-    });
+    return isEnumOfFloats(key) ?
```

Comment:
So far I've only seen enums that are either of strings or floats. Both of these map to `comparable` elm types, which can be used as Dict keys. We will still create the union types for these, but when used in a context that required comparable, we'll just set a `String` or `Float`